react-validated-input/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { FcCheckmark } from 'react-icons/fc';
+import { FaTimes } from 'react-icons/fa';
+
+export function ValidatedInput() {
+  const [charCount, setCharCount] = useState(0);
+  const label = <label htmlFor="password">Password:</label>;
+  const input = (
+    <input type="text" id="password" onChange={handleCharChange}></input>
+  );
+  /*   const passwordRegEx = /^(?=.*[a-zA-Z])(?=.*d)(?=.*[!#$%&?]).{8,}$/;
+  const formatResult = passwordRegEx.test(); */
+  let message = '';
+  let passwordFormat = false;
+  if (charCount === 0) {
+    message = 'A password is required.';
+    passwordFormat = false;
+  } else if (charCount < 8) {
+    message = 'Your password is too short.';
+    passwordFormat = false;
+  } else {
+    message = '';
+    passwordFormat = true;
+  }
+  const pMessage = (
+    <p className="message" style={{ textAlign: 'left' }}>
+      {message}
+    </p>
+  );
+  function handleCharChange() {
+    setCharCount(charCount + 1);
```

Comment:
This doesn't work. If you type 8 characters your password will transition to valid, but if you then delete some characters, it will remain valid.

Your state should be the actual password, not the number of characters. Although it was a clever idea!

Note that the onChange handler will be passed an `event`. You can look at `event.target.value` to get the current value of the password.